src/store/index.js

```diff
@@ -55,8 +61,93 @@ export default new Vuex.Store({
+        .createUserWithEmailAndPassword(email, password);
+      const newUser = { id: user.uid };
+      commit('setUser', newUser);
+    },
+    async signUserIn({ commit }, { email, password }) {
+      const { user } = await firebase.auth()
+        .signInWithEmailAndPassword(email, password);
+      commit('setUser', { id: user.uid });
+    },
+    logout({ commit }) {
+      commit('resetStore');
+      firebase.auth().signOut();
+    },
+    autoSignIn(store, { uid }) {
+      store.commit('setUser', { id: uid });
+    },
+    async createTaskList({ commit, state }, listTitle) {
+      const userId = state.user.id;
+      const data = await firebase.database().ref(`lists/${userId}`).push(listTitle);
+      commit('addNewList', { id: data.key, title: listTitle });
+    },
+    async loadLists({ commit, state }) {
+      const userId = state.user.id;
+      const lists = await firebase.database().ref(`lists/${userId}`)
+        .once('value').then((snapshot) => snapshot.val()) || {};
+      const listsArr = Object.entries(lists);
+      listsArr.forEach((list) => {
+        commit('addNewList', { id: list[0], title: list[1] });
+      });
+    },
+    async removeList({ commit, state }, listId) {
+      const userId = state.user.id;
+      await firebase.database().ref(`lists/${userId}/${listId}`).set(null);
+      await firebase.database().ref(`tasks/${userId}/${listId}`).set(null);
+      commit('removeList', listId);
+    },
+    async createTask({ commit, state }, { taskText, listId }) {
+      const userId = state.user.id;
+      const data = await firebase.database().ref(`tasks/${userId}/${listId}`)
+        .push({ state: false, taskText });
+      commit('addTask', {
+        taskText,
+        listId,
+        taskId: data.key,
+        taskState: false,
+      });
+    },
+    async removeTask({ commit, state }, { taskId, listId }) {
+      const userId = state.user.id;
+      await firebase.database().ref(`tasks/${userId}/${listId}/${taskId}`).set(null);
+      commit('removeTask', { taskId, listId });
+    },
+    async updateTaskState({ commit, state }, { taskId, listId }) {
+      const userId = state.user.id;
+      const task = state.tasks[listId].find((item) => item.taskId === taskId);
+      await firebase.database()
+        .ref(`tasks/${userId}/${listId}/${taskId}/state`).set(!task.taskState);
+      commit('updateTaskState', { taskId, listId });
+    },
+    async loadTasks({ commit, state }) {
+      const userId = state.user.id;
+      const tasks = await firebase.database().ref(`tasks/${userId}`)
+        .once('value').then((snapshot) => snapshot.val()) || {};
+      const tasksArr = Object.entries(tasks);
+      tasksArr.forEach((taskList) => {
+        Object.entries(taskList[1]).forEach((item) => {
+          commit('addTask', {
+            taskText: item[1].taskText,
+            listId: taskList[0],
+            taskId: item[0],
+            taskState: item[1].state,
+          });
```

Comment:
It is a bad idea to call mutation in loop, will be better to create an array with whole date and sent it to state. In current implementation you also need to add separate mutation for this ('setLoadedTasks' e.g.).